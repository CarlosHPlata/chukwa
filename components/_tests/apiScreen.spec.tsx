import { render } from "@testing-library/react-native";
import { Text } from "react-native";
import ApiScreen from "../apiScreen";

describe("ApiScreen", () => {
  let consoleErrorSpy: jest.SpyInstance;

  beforeAll(() => {
    consoleErrorSpy = jest.spyOn(console, 'error').mockImplementation(() => {});
  });

  afterAll(() => {
    consoleErrorSpy.mockRestore();
  });

  it("renders correctly", () => {
    const { toJSON } = render(
      <ApiScreen isLoading={false} error={undefined}>
        <></>
      </ApiScreen>
    );
    expect(toJSON()).toMatchSnapshot();
  });
  
  it("renders loading state", () => {
    const { getByTestId } = render(
      <ApiScreen isLoading={true} error={undefined}>
        <></>
      </ApiScreen>
    );

    expect(getByTestId("api-screen:activity-indicator")).toBeTruthy();
  });

  it("renders error state", () => {
    const error = new Error("Test error");
    const { getByText } = render(
      <ApiScreen isLoading={false} error={error}>
        <></>
      </ApiScreen>
    );
    expect(getByText("Test error")).toBeTruthy();
  });

  it("renders children when not loading and no error", () => {
    const { getByText } = render(
      <ApiScreen isLoading={false} error={undefined}>
        <Text>Content</Text>
      </ApiScreen>
    );

    expect(getByText("Content")).toBeTruthy();
  });
});
