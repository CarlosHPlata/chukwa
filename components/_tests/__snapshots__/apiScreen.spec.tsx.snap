// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApiScreen renders correctly 1`] = `null`;
